File: frontend/types/trip.ts
interface Trip {
  id: string
  destination: string
  days: number
  budget: number
  styles: string[]
  daily_budget: number
  category: string
  recommendation?: string
  pending: boolean
  created_at: string
  updated_at: string
  row_num: number
}

interface TripRequest {
  destination: string
  budget: number
  days: number
  styles: string[]
}

interface TripResponse {
  success: boolean
  data?: Trip
}

interface TripSearchRequest {
  query: string
  filter?: { destination: boolean, style: boolean }
  page?: { index: number, size: number }
}

interface TripSearchResponse {
  data: Trip[]
  total: number
}

type TripStatusResponse = {
  success: boolean
  data?: {
    id: string
    pending: true
    message: string
  } | {
    id: string
    pending: false
    recommendation: string | null
  }
}

interface LoginRequest {
  email: string
  password: string
}

interface RegisterRequest {
  name: string
  email: string
  password: string
}

interface UserProfile {
  name: string
  email: string
  picture?: string
}

export type {
  Trip, TripRequest, TripResponse, TripSearchRequest, TripSearchResponse, TripStatusResponse,
  RegisterRequest, LoginRequest,
  UserProfile
};

File: backend/app/schemas/trip.py
from enum import StrEnum

from pydantic import BaseModel, ConfigDict


class TripCategory(StrEnum):
    BACKPACKER = "backpacker"
    STANDARD = "standard"
    LUXURY = "luxury"


class TripCreateRequest(BaseModel):
    destination: str
    days: int
    budget: float
    styles: list[str] | None = None


class TripCreateResponse(BaseModel):
    model_config = ConfigDict(use_enum_values=True)

    id: str
    destination: str
    days: int
    budget: float
    category: TripCategory
    styles: list[str] | None = None


class TripFilterRequest(BaseModel):
    destination: str | None = None
    style: str | None = None


class TripResponse(BaseModel):
    model_config = ConfigDict(use_enum_values=True)

    id: str
    account_id: str
    destination: str
    days: int
    budget: float
    daily_budget: float
    category: TripCategory
    styles: list[str]
    created_at: str
    pending: bool | None = False
    error: str | None = None
    updated_at: str | None = None
    row_num: int | None = 0


class TripDetailResponse(BaseModel):
    model_config = ConfigDict(use_enum_values=True)

    id: str
    account_id: str
    destination: str
    days: int
    budget: float
    daily_budget: float
    category: TripCategory
    styles: list[str]
    created_at: str
    recommendation: str | None
    pending: bool | None = False
    error: str | None = None
    updated_at: str | None = None
    row_num: int | None = 0


class TripStatusResponse(BaseModel):
    id: str
    pending: bool
    recommendation: str | None = None
    message: str | None = None
